src/main/java/org/dataloader/DataLoaderRegistry.java

```diff
@@ -142,7 +144,7 @@ private static DataLoaderOptions setInInstrumentation(DataLoaderOptions options,
      */
     public DataLoaderRegistry register(DataLoader<?, ?> dataLoader) {
         String name = dataLoader.getName();
-        assertState(name != null, () -> "The DataLoader must have a non null name");
+        Objects.requireNonNull(name, "The DataLoader must have a non null name");
```

Comment:
The tooling understands Objects.requireNonNull but not out nonNull method

Reply:
You can teach tooling about this custom assertion method.

Spring Framework does that with [a custom `@Contract` annotation](https://docs.spring.io/spring-framework/reference/7.0-SNAPSHOT/core/null-safety.html#_configuration), used on [the relevant methods](https://github.com/spring-projects/spring-framework/blob/main/spring-core/src/main/java/org/springframework/util/Assert.java#L77-L78).

In a nutshell, two possibilities:
1. use methods like `Objects.requireNonNull`, already known by tooling
2. OR create a custom contract annotation, annotate the relevant methods in your codebase with it and [teach NullAway about this annotation](https://github.com/spring-projects/spring-framework/blob/0c6a26a381505479ae62a67eb018a5e41447bbb7/gradle/spring-module.gradle#L133).

Reply:
Thanks - for the record the reason we have a custom Assert class is that back in Java 6 days - Objects.requireNonNull() didnt exist and we wanted near zero dependendencies - so we made out own

Same in graphql-java btw